src/renderer/components/ExternalPlayerSettings.vue

```diff
@@ -137,13 +142,67 @@ function updateExternalPlayerIgnoreDefaultArgs(value) {
   store.dispatch('updateExternalPlayerIgnoreDefaultArgs', value)
 }
 
+/**
+ * Replace any %VAR% tokens (like %Program Files%) in a path string with the corresponding process.env.VAR value.
+ * @param {string} str
+ * @returns {string}
+ */
+function expandEnvVars(str) {
+  return str.replaceAll(/%([^%]+)%/g, (_, name) => process.env[name] || '')
+}
+
+/**
+ * Detect os
+ * @returns {'Windows'|'macOS'|'Linux'|'Unknown'}
+ */
+function detectOS() {
```

Comment:
We have a build variable that contains the os that should be used instead of this custom function.

process.platform === 'darwin' // for macos
process.platform === 'win32' // for windows
process.platform === 'linux' // for linux